client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -663,6 +664,26 @@ private Boolean handleScaleResponse(Stream stream, ScaleResponse response, long
         });
     }
 
+    @Override
+    public CompletableFuture<Boolean> isStreamCutValid(final StreamCut streamCut) {
+        Exceptions.checkNotClosed(closed.get(), this);
+        long traceId = LoggerHelpers.traceEnter(log, "isStreamCutValid", streamCut);
+
+        final CompletableFuture<StreamCutValidityResponse> result = this.retryConfig.runAsync(() -> {
+            RPCAsyncCallback<StreamCutValidityResponse> callback = new RPCAsyncCallback<>(traceId, "isStreamCutValid");
+            client.isStreamCutValid(io.pravega.controller.stream.api.grpc.v1.Controller.StreamCut.newBuilder().putAllCut(getStreamCutMap(streamCut)).build(),
+                    callback);
+            return callback.getFuture();
+        }, this.executor);
+        return result.thenApply(StreamCutValidityResponse::getResponse)
+                .whenComplete((x, e) -> {
+                    if (e != null) {
+                        log.warn("isStreamCutValid failed: ", e);
```

Comment:
Why bother to log this exception here when it would bubble up to the caller anyway?